engine/table/src/main/java/io/deephaven/engine/table/impl/sources/ring/AbstractRingChunkSource.java

```diff
@@ -73,7 +73,13 @@ public AbstractRingChunkSource(@NotNull Class<T> type, int capacity) {
         }
         this.capacity = capacity;
         // noinspection unchecked
-        ring = (ARRAY) Array.newInstance(type, capacity);
+        ring = type.isArray()
```

Comment:
I think the wider issue here is that this code is using reflection to determine the array type instead of using inheritance (and using `Object[]` to hold non-primitive values).

Reply:
Are you certain you shouldn't be using `Object[]` in more cases?

Reply:
Never mind, we figured out that we were dealing with weird upstream data.